openmrs/src/test/java/org/motechproject/openmrs/it/MRSTasksIntegrationBundleIT.java

```diff
@@ -0,0 +1,481 @@
+
+    @Inject
+    private OpenMRSProviderService providerService;
+
+    @Inject
+    private OpenMRSPersonService personService;
+
+    @Inject
+    private TaskActivitiesDataService taskActivitiesDataService;
+
+    private Patient createdPatient;
+    private Encounter createdEncounter;
+    private Provider createdProvider;
+    private EncounterType createdEncounterType;
+
+    private static final String OPENMRS_CHANNEL_NAME = "org.motechproject.openmrs";
+    private static final String OPENMRS_MODULE_NAME = "openMRS";
+    private static final String MDS_CHANNEL_NAME = "org.motechproject.motech-platform-dataservices-entities";
+    private static final String VERSION = "0.29.0.SNAPSHOT";
+    private static final String TEST_INTERFACE = "org.motechproject.openmrs.tasks.OpenMRSActionProxyService";
+    private static final String TRIGGER_SUBJECT = "mds.crud.serverconfig.SettingsRecord.CREATE";
+    private static final String MOTECH_ID = "654";
+    private static final String TEST_MOTECH_ID = "333";
+
+    private static final Integer MAX_RETRIES_BEFORE_FAIL = 20;
+    private static final Integer WAIT_TIME = 2000;
+
+    @Override
+    protected Collection<String> getAdditionalTestDependencies() {
+        return Arrays.asList(
+                "org.motechproject:motech-tasks-test-utils",
+                "org.motechproject:motech-tasks",
+                "commons-beanutils:commons-beanutils",
+                "commons-fileupload:commons-fileupload",
+                "org.motechproject:motech-platform-web-security",
+                "org.motechproject:motech-platform-server-bundle",
+                "org.openid4java:com.springsource.org.openid4java",
+                "net.sourceforge.nekohtml:com.springsource.org.cyberneko.html",
+                "org.springframework.security:spring-security-openid"
+        );
+    }
+
+    @BeforeClass
+    public static void initialize() throws IOException, InterruptedException {
+        createAdminUser();
+        login();
+    }
+
+    @Before
+    public void setUp() throws IOException, InterruptedException, ParseException, ConceptNameAlreadyInUseException {
+        openMRSTasksNotifier = (OpenMRSTasksNotifier) ServiceRetriever
+                .getWebAppContext(bundleContext, OPENMRS_CHANNEL_NAME)
+                .getBean("openMrsTasksNotifier");
+
+        setUpSecurityContext("motech", "motech", "manageTasks", "manageOpenMRS");
+
+        waitForChannel(OPENMRS_CHANNEL_NAME);
+        waitForChannel(MDS_CHANNEL_NAME);
+
+        createPatientTestData();
+        createEncounterTestData();
+    }
+
+    @Test
+    public void testOpenMRSPatientDataSourceAndCreatePatientAction() throws InterruptedException, IOException, PatientNotFoundException {
+        Long taskID = createPatientTestTask();
+
+        activateTrigger();
+
+        // Give Tasks some time to process
+        assertTrue(waitForTaskExecution(taskID));
+
+        deleteTask(taskID);
```

Comment:
there is a problem with deleting task, sometimes it is invoked too early, please remove deleting task, the whole motech db is dropped before OPENMRSITs